packages/core/src/extension/extension-service.ts

```diff
@@ -34,10 +35,23 @@ export default class ExtensionService {
 
         this._eventListener.listen();
         this._commandPoster.setContext({ extensionId });
-        this._internalCommandPoster.post({
-            type: ExtensionInternalCommandType.ResizeIframe,
-            payload: { extensionId },
-        });
+
+        try {
+            await this._internalCommandPoster.post(
+                {
+                    type: ExtensionInternalCommandType.ResizeIframe,
+                    payload: { extensionId },
+                },
+                {
+                    successType: ExtensionInternalEventType.ExtensionReady,
+                    errorType: ExtensionInternalEventType.ExtensionFailed,
+                },
+            );
+        } catch (event) {
+            if (this._isExtensionFailedEvent(event)) {
+                throw new Error('Extension failed to initialize within 60 seconds');
```

Comment:
When the host side fails to get the`iframeResizer`'s `init` message from the client side  within 60 seconds, it will send back `ExtensionInternalEventType.ExtensionFailed` and trigger this error.

Reply:
Do we want to proceed this further, such as refreshing the page?

Reply:
Can we just return to the client that something failed so they can do a retry at their end?

Reply:
I have changed the error message to `Extension failed to load within 60 seconds; please reload and try again.`. A dev should be able to know what to do next. 😆